src/hotspot/share/classfile/symbolTable.cpp

```diff
@@ -72,12 +72,14 @@ bool SymbolTable::_needs_rehashing = false;
 bool SymbolTable::_lookup_shared_first = false;
 static juint murmur_seed = 0;
 
+#ifdef ASSERT
```

Comment:
Any reason why wrapping hash_symbol with #ifdef ASSERT?

Reply:
My gcc always complains unused function hash_symbol when building release. I noticed this function is only used for debug code which guarded by assert. So I guard hash_symbol with assert as well.

Reply:
Ok, thanks. In JDK 11, hash_symbol(const char* s, int len, bool useAlt) is only used by the following in CopyToArchive:

    assert(fixed_hash == hash_symbol((const char*)sym->bytes(), sym->utf8_length(), false),
           "must not rehash during dumping");